libscript/src/script-instance.cpp

```diff
@@ -1123,27 +1144,27 @@ __MCScriptResolveForeignFunctionBindingForObjC(MCScriptInstanceRef p_instance,
                 /* ERROR: OOM */
                 t_valid = false;
             }
-            t_cif = static_cast<ffi_cif *>(*t_layout_cif);
-            t_cif_arg_types = reinterpret_cast<ffi_type **>(t_cif + 1);
-        }
-        
-        unsigned int t_user_arg_index = 0;
-        unsigned int t_arg_index = 0;
-        
-        /* The first argument is always of pointer type - the object ptr. This
-         * is only present in the user signature if it is an instance method.*/
-        t_cif_arg_types[t_arg_index] = &ffi_type_pointer;
-        t_arg_index += 1;
-        if (!t_is_class)
-        {
-            t_user_arg_index += 1;
+            else
+            {
+                t_cif = static_cast<ffi_cif *>(*t_layout_cif);
+                t_cif_arg_types = reinterpret_cast<ffi_type **>(t_cif + 1);
+                
+                /* The first argument is always of pointer type - the object ptr. This
+                 * is only present in the user signature if it is an instance method.*/
+                t_cif_arg_types[t_arg_index] = &ffi_type_pointer;
```

Comment:
Note that this code has been moved because my changes above exposed a crash here when `t_valid` as false